services-bytebuf-codec/src/main/java/io/scalecube/services/transport/api/DefaultHeadersCodec.java

```diff
@@ -0,0 +1,84 @@
+package io.scalecube.services.transport.api;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/** Simple binary codec for headers service message. */
+public class DefaultHeadersCodec implements HeadersCodec {
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public String contentType() {
+    return "application/octet-stream";
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void encode(OutputStream stream, Map<String, String> headers) throws IOException {
+    if (headers.isEmpty()) {
```

Comment:
Im in doubt do we need conditions :
[32](https://github.com/scalecube/scalecube-services/pull/618/files#diff-0ff374c48aefdf1e324c0c0ac0292969R32) :
```
if (headers.isEmpty()) {
  ...
}
```
and at [52](https://github.com/scalecube/scalecube-services/pull/618/files#diff-0ff374c48aefdf1e324c0c0ac0292969R52):
```
if (stream.available() < 1) {
  ...
}
```

If  headers are empty then u will write at [32](https://github.com/scalecube/scalecube-services/pull/618/files#diff-0ff374c48aefdf1e324c0c0ac0292969R32)  size `0`. And then at [55](https://github.com/scalecube/scalecube-services/pull/618/files#diff-0ff374c48aefdf1e324c0c0ac0292969R55)  u just return hashmap with size `0`

Reply:
why write if you can not write?